repobee_sanitizer/_sanitize_repo.py

```diff
@@ -102,6 +107,44 @@ def create_extension_command(self) -> plug.ExtensionCommand:
         )
 
 
+def _parse_file_list(file_list) -> List[pathlib.Path]:
+    return [
+        p.strip()
+        for p in file_list.read_text(encoding=_ASSUMED_ENCODING)
+        .strip()
+        .split("\n")
+    ]
+
+
+def _discover_dirty_files(repo_root) -> List[pathlib.Path]:
+    """
+    Returns:
+        A list of relative file paths for files that need to be sanitized.
+    """
+    return [
+        file
+        for file in list(repo_root.rglob("*"))
+        if ".git" not in str(file) and file.is_file() and _file_is_dirty(file)
+    ]
+
+
+def _file_is_dirty(file: pathlib.Path) -> bool:
+    """if a file has ANY block, pass it forward, its syntax will then be
+    checked when sanitized.
+    """
+
+    content = file.read_text(encoding=_ASSUMED_ENCODING).split("\n")
+    for line in content:
+        for block in [
+            "REPOBEE-SANITIZER-BLOCK",
+            "REPOBEE-SANITIZER-REPLACE-WITH",
+            "REPOBEE-SANITIZER-END",
```

Comment:
If we only check start blocks for example, if there is a file with only an end block, it will not be sent onward, and then not sanitized properly

Reply:
Correct, we should check for any block

Reply:
However, here you're doing a big no-no: writing out the markers as strings. Use the defined constants instead.
```suggestion
            _sanitize.START_BLOCK,
            _sanitize.REPLACE_WITH,
            _sanitize.END_BLOCK,
```

It would probably make sense to put all markers in a list at the same place where they are defined, so we don't run the risk of missing a new marker if added.